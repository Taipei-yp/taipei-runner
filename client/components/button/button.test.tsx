import React from "react";
import TestRenderer, { act } from "react-test-renderer";
import { Button } from ".";

describe("Component Button", () => {
  let onClickCall = false;
  const clickFunc = () => {
    onClickCall = true;
  };
  const button = TestRenderer.create(
    <Button
      viewType="secondary"
      className="testclass"
      type="submit"
      onClick={clickFunc}
    >
      test button
    </Button>,
  );

  it("correct render", () => {
    expect(button.toJSON()).toMatchSnapshot();
  });
  it("correct props", () => {
    expect(button.root.props.viewType).toBe("secondary");
    expect(button.root.props.className).toBe("testclass");
    expect(button.root.props.type).toBe("submit");
  });
  it("onclick work", () => {
    act(button.root.props.onClick);
    expect(onClickCall).toBe(true);
  });
});
