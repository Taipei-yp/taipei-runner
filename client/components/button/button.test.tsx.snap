// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button correct render 1`] = `
<button
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  onClick={[Function]}
  type="submit"
>
  test button
</button>
`;
